### PR TITLE
Update OpenShift template

### PR DESCRIPTION
**Pull request by rsnyman**
- Update the OpenShift template YAML
- Removed Dynaconf, cause it was just messing us around

### CODE REVIEW

File: ibutsu-server-openshift.yaml [hunk 1]
```diff
@@ -285,30 +211,30 @@ objects:
         - env:
           - name: APP_CONFIG
             value: config.py
-          - name: DYNACONF_HAS_FRONTEND
+          - name: HAS_FRONTEND
             value: "false"
-          - name: DYNACONF_HOST
+          - name: POSTGRESQL_HOST
             value: mongodb.ibutsu-server.svc
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:217]: Value here looks like it needs updating

File: ibutsu-server-openshift.yaml [hunk 2]
```diff
@@ -285,30 +211,30 @@ objects:
...
-          - name: DYNACONF_PORT
+          - name: POSTGRESQL_PORT
             value: "27017"
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:219]: Port should be update to "5432"?

File: ibutsu-server-openshift.yaml [hunk 3]
```diff
@@ -285,30 +211,30 @@ objects:
...
-          - name: DYNACONF_USERNAME
+          - name: POSTGRESQL_USER
             valueFrom:
               secretKeyRef:
                 key: database-user
                 name: mongodb
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:224]:
```suggestion
                name: postgresql-db
```

File: ibutsu-server-openshift.yaml [hunk 4]
```diff
@@ -285,30 +211,30 @@ objects:
...
-          - name: DYNACONF_PASSWORD
+          - name: POSTGRESQL_PASSWORD
             valueFrom:
               secretKeyRef:
                 key: database-password
                 name: mongodb
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:229]:
```suggestion
                name: postgresql-db
```

File: ibutsu-server-openshift.yaml [hunk 5]
```diff
@@ -285,30 +211,30 @@ objects:
...
-          - name: DYNACONF_DATABASE
+          - name: POSTGRESQL_DATABASE
             valueFrom:
               secretKeyRef:
                 key: database-name
                 name: mongodb
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:234]:
```suggestion
                name: postgresql-db
```

File: ibutsu-server-openshift.yaml [hunk 6]
```diff
@@ -484,100 +602,295 @@ objects:
...
-            failureThreshold: 3
-            initialDelaySeconds: 30
-            periodSeconds: 10
-            successThreshold: 1
-            tcpSocket:
-              port: 27017
-            timeoutSeconds: 1
-          name: mongodb
+                name: postgresql
+          - name: DYNACONF_CELERY_BROKER_URL
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:729]: Since we're abandoning dynaconf, maybe this should change too

File: ibutsu-server-openshift.yaml [hunk 7]
```diff
@@ -484,100 +602,295 @@ objects:
...
+            value: redis://:${REDIS_PASSWORD}@redis.${NAMESPACE}.svc
+          - name: DYNACONF_CELERY_RESULT_BACKEND
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:731]: here too

File: ibutsu-server-openshift.yaml [hunk 8]
```diff
@@ -484,100 +602,295 @@ objects:
...
+  spec:
+    ports:
+    - name: postgresql
+      port: 5432
+      protocol: TCP
+      targetPort: 5432
+    selector:
+      deploymentConfig: postgresql
+# -----------------------------------------------
+- kind: PersistentVolumeClaim
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:866]: Do we need another for the backup volume claim?

File: ibutsu-server-openshift.yaml [hunk 9]
```diff
@@ -1024,3 +1139,12 @@ parameters:
   displayName: Backend Route
   description: The URL of the backend of the Ibutsu server
   value: ibutsu-api.example.com
+- name: BACKUP_VOLUME_CLAIM
+  displayName: Backup volumn claim
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:1143]:
```suggestion
  displayName: Backup volume claim
```

**Review by john-dupuy — CHANGES_REQUESTED** [review]
Awesome :D, I noticed a few small typos/renaming things

**Comment by john-dupuy** [comment]
From the travis build looks like there are a couple of references to dynaconf that need removing:
```
  File "/home/travis/build/ibutsu/ibutsu-server/backend/ibutsu_server/config.py", line 1, in <module>
    from dynaconf import Dynaconf
ModuleNotFoundError: No module named 'dynaconf'
```

File: ibutsu-server-openshift.yaml [hunk 10]
```diff
@@ -262,6 +106,88 @@ objects:
...
+      git:
+        uri: ${IBUTSU_REPO_URL}
+        ref: ${IBUTSU_REPO_BRANCH}
+      contextDir: backend
+    strategy:
+      type: Source
+      sourceStrategy:
+        from:
+          kind: ImageStreamTag
+          name: python:3.6
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:174]: possible to use python:3.7 or 3.8?

File: ibutsu-server-openshift.yaml [hunk 11]
```diff
@@ -1024,3 +1154,15 @@ parameters:
...
+  value: database-backup
+- name: BACKUP_STORAGE
+  displayName: Backup storage
+  value: 30Gi
+- name: BACKUP_KEEP
+  displayName: Number of backups to keep
+  value: '5'
+- name: BACKUP_SCHEDULE
+  displayName: Cron-like schedule to run backup
+  value: '50 * * * *'
```
Comment (john-dupuy) [inline ibutsu-server-openshift.yaml:1168]: Will backing up this often impact performance?

File: ibutsu-server-openshift.yaml (hunk 10 above)
Comment (rsnyman) [reply ibutsu-server-openshift.yaml:174]: Hrm, let me see what's available in OCPv4

File: ibutsu-server-openshift.yaml (hunk 11 above)
Comment (rsnyman) [reply ibutsu-server-openshift.yaml:1168]: Heh, whoops, that's from the example. I need to fix that.

File: ibutsu-server-openshift.yaml (hunk 6 above)
Comment (john-dupuy) [reply ibutsu-server-openshift.yaml:729]:
```suggestion
          - name: CELERY_BROKER_URL
```

File: ibutsu-server-openshift.yaml (hunk 7 above)
Comment (john-dupuy) [reply ibutsu-server-openshift.yaml:731]:
```suggestion
          - name: CELERY_RESULT_BACKEND
```

File: ibutsu-server-openshift.yaml (hunk 10 above)
Comment (rsnyman) [reply ibutsu-server-openshift.yaml:174]: OCPv4 only has 3.6 in the catalog. We can try 3.8 and see if it pulls it in, but I don't know that it will.

**Review by john-dupuy — APPROVED** [review]

**Pull request merged**